hazelcast/src/main/java/com/hazelcast/map/impl/MapEntries.java

```diff
@@ -85,4 +77,28 @@ public int getFactoryId() {
     public int getId() {
         return MapDataSerializerHook.ENTRY_SET;
     }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        int size = entries.size();
+        out.writeInt(size);
+
+        if (size > 0) {
+            for (Map.Entry<Data, Data> o : entries) {
+                out.writeData(o.getKey());
+                out.writeData(o.getValue());
+            }
+        }
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        int size = in.readInt();
+        for (int i = 0; i < size; i++) {
```

Comment:
I know. The problem is that the no arg constructor; needed for serialization, is also used for regular calls. And for regular calls, you need to have an initialized collections; hence it is made in the constructor. I'm not very happy with it, but don't see a clear way out which some ugly code. What is your suggestion.